Добавлена первая ЛР

Iutin12 opened the pull request.

Iutin12 committed:
Добавлена первая ЛР

vladushka2000 commented on LR1.ipynb:
```diff
@@ -0,0 +1,194 @@
+    "            print('Экземпляр уже создан')\n",
+    "            return\n",
+    "\n",
+    "        self.lock = threading.Lock()\n",
```
Вынести локи в переменные класса. Должно быть 2 штуки их

vladushka2000 commented on LR1.ipynb:
```diff
@@ -0,0 +1,194 @@
+    "        now = datetime.datetime.now()\n",
+    "        return f\"DP.P1.{now.strftime('%Y-%m-%d.%H-%M-%S')}.log\"\n",
+    "\n",
+    "    def log(self, level, message):\n",
```
Сделать методом protected

vladushka2000 commented on LR1.ipynb:
```diff
@@ -0,0 +1,194 @@
+    "import os\n",
+    "\n",
+    "\n",
+    "class Logger:\n",
```
Нету ни одной докстринги. Опиши, что делает класс и каждый метод с тайп хинтами

vladushka2000 commented on LR1.ipynb:
```diff
@@ -0,0 +1,194 @@
+    "            cls._instance.__initialized = False\n",
+    "        return cls._instance\n",
+    "\n",
+    "    def __init__(self, directory_path):\n",
```
Ты тут по сути ограничиваешься в себя создании будущих экземпляров класса. Убери проверку и переменную initialized

vladushka2000 commented on LR1.ipynb:
```diff
@@ -0,0 +1,194 @@
+    "            with open(self.file_name, 'w') as file:\n",
+    "                pass\n",
+    "\n",
+    "    def get_log_file_name(self) -> str:\n",
```
Сделать protected

vladushka2000 commented on LR1.ipynb:
```diff
@@ -0,0 +1,194 @@
+    "    logger.info(\"Программа запущена успешно\")\n",
+    "\n",
+    "\n",
+    "    logger = Logger(save_folder_path)\n",
```
Ты назначил все в одну и ту же переменную. Попробуй создать новую с новым логгером

Rognak commented on LR1.ipynb:
```diff
@@ -0,0 +1,194 @@
+{
```
Замечания:
1) Отсутствуют аннотации типов
2) commit'ы оформлены не в соответствии с Conventional Commits
3) get_log_file_name можно сделать как @property (log_file_name)

Rognak replied:
+ Исправить замечания от Влада

Iutin12 committed:
fix(logger): исправлены замечания

vladushka2000 commented on LR1.ipynb:
```diff
@@ -0,0 +1,204 @@
+    "\n",
+    "\n",
+    "class Logger:\n",
+    "    \"\"\"КЛасс для создания логгера\"\"\"\n",
```
Ты должен описывать класс как сущность, а не его предназначение. Класс логгера

vladushka2000 commented on LR1.ipynb:
```diff
@@ -0,0 +1,204 @@
+    "    \"\"\"КЛасс для создания логгера\"\"\"\n",
+    "\n",
+    "    def __new__(cls, *args):\n",
+    "        \"\"\"Создает экземпляр\"\"\"\n",
```
И где тут синглтон?

vladushka2000 commented on LR1.ipynb:
```diff
@@ -0,0 +1,204 @@
+    "        return instance\n",
+    "\n",
+    "    def __init__(self, directory_path: str):\n",
+    "        \"\"\"Инициализатор, где создается путь для записи файла\"\"\"\n",
```
Описание действия метода. "Инициализировать переменные". Нету описания аргументов

vladushka2000 commented on LR1.ipynb:
```diff
@@ -0,0 +1,204 @@
+    "    def __init__(self, directory_path: str):\n",
+    "        \"\"\"Инициализатор, где создается путь для записи файла\"\"\"\n",
+    "\n",
+    "        self._lock = threading.Lock()\n",
```
ты так и не исправил объект lock. Должно быть 2 лока на уровне КЛАССА, не экземпляра

vladushka2000 commented on LR1.ipynb:
```diff
@@ -0,0 +1,204 @@
+    "            with open(self._file, 'a', encoding='utf-8') as file:\n",
+    "                file.write(log_message)\n",
+    "\n",
+    "    \"\"\"Дальше идет запись сообщений с разным уровнем важности\"\"\"\n",
```
опиши каждый метод нормальной докстрингой с описанием аргументов

vladushka2000 commented on LR1.ipynb:
```diff
@@ -0,0 +1,204 @@
+    "\n",
+    "    logger1 = Logger(save_folder_path)\n",
+    "    logger1.trace(\"Начало работы программы1\")\n",
+    "    logger1.info(\"Программа запущена успешно1\")\n",
```
сделай проверку на идентичность объектов logger1 и logger2

vladushka2000 commented on LR1.ipynb:
```diff
@@ -0,0 +1,204 @@
+    "        now = datetime.datetime.now()\n",
+    "        return f\"DP.P1.{now.strftime('%Y-%m-%d.%H-%M-%S')}.log\"\n",
+    "\n",
+    "    def _log(self, level: str, message: str) -> None:\n",
```
неполная докстринга. Нету описания аргументов

vladushka2000 commented on LR1.ipynb:
```diff
@@ -0,0 +1,204 @@
+    "\n",
+    "        self._lock = threading.Lock()\n",
+    "        self._file_name = self._get_log_file_name()\n",
+    "        self._file = pathlib.Path(directory_path) / self._file_name\n",
```
это не файл, а путь к файлу. Исправь название поля

vladushka2000 commented:
Ну и раз уже проверки доходят до стиля коммитов, то от меня замечание по их содержанию. "Исправлены замечания" - не информативно.

fix(logger): Исправлена блокировка потоков

Такой коммит в 100 раз информативнее. Лучше куча коммитов с исправлением каких-то частей, а не один большой ком из всего-всего. Тем более потом просто можно сделать сквош для красивой истории

Iutin12 committed:
fix(logger): Исправлена блокировка потоков и прочие комментарии с ошибками

ДЕлать много маленьких комментариев пока не было времени, поэтому одним большим

vladushka2000 commented on LR1.ipynb:
```diff
@@ -0,0 +1,222 @@
+    "    def __new__(cls, *args, **kwargs):\n",
+    "        \"\"\"Создает единственный экземпляр.\"\"\"\n",
+    "        if cls._instance is None:\n",
+    "            cls._instance = super(Logger, cls).__new__(cls)\n",
```
нету вызова лока на создание логгера

Iutin12 committed:
fix(logging): добавлен вызов лока

vladushka2000 commented on LR1.ipynb:
```diff
@@ -0,0 +1,223 @@
+    "        Args:\n",
+    "            directory_path (str): Путь к директории, где будет создан файл для записи логов.\n",
+    "        \"\"\"\n",
+    "        with Logger._init_lock:\n",
```
убираешь вот это и я засчитываю

Iutin12 committed:
fix(logger): убран комментарий
